### PR TITLE
Added a troubleshooting section for user guide.

### PR DESCRIPTION
**Pull request by gyliu513**
Fixed https://github.com/kubernetes-sigs/federation-v2/issues/350

/cc @danehans @marun @font

### CODE REVIEW

File: docs/userguide.md [hunk 1]
```diff
@@ -305,6 +305,15 @@ To cleanup the example simply delete the namespace:
 kubectl delete ns test-namespace
 ```
 
+### Troulbeshooting
```
Comment (marun) [inline docs/userguide.md]: s/Troulbeshooting/Troubleshooting/

File: docs/userguide.md [hunk 2]
```diff
@@ -305,6 +305,15 @@ To cleanup the example simply delete the namespace:
...
+
+If you found the federated resources were not propagated to the member clusters, you can always
+use the following command to get some failure reason.
+
+```bash
+kubectl describe <federate resource type> -n test-namespace
```
Comment (marun) [inline docs/userguide.md]: What does `<federate resource type>` refers to?  And what information should the user expect to see?  I was under the impression that determining propagation errors would require inspecting the log of the controller manager, but maybe I'm forgetting something.

Comment (gyliu513) [reply docs/userguide.md]: Done, I updated `<federate resource type>` to `<federate CRD>`, also add a step for how to check controller log.

File: docs/userguide.md [hunk 3]
```diff
@@ -305,6 +305,27 @@ To cleanup the example simply delete the namespace:
 kubectl delete ns test-namespace
 ```
 
+### Troubleshooting
+
+If you found the federated resources were not propagated to the member clusters, you can always
+use the following command to get some failure reason by checkig the `Events`.
```
Comment (marun) [inline docs/userguide.md]: s/to get some failure reason by checkig the Events/to view Events which may aid in diagnosing the problem/

File: docs/userguide.md [hunk 4]
```diff
@@ -305,6 +305,27 @@ To cleanup the example simply delete the namespace:
...
+
+```bash
+kubectl describe <federate CRD> <CR> -n test-namespace
```
Comment (marun) [inline docs/userguide.md]: s/federate/federated/
s/CR/CR name/

File: docs/userguide.md [hunk 5]
```diff
@@ -305,6 +305,27 @@ To cleanup the example simply delete the namespace:
 kubectl delete ns test-namespace
 ```
 
+### Troubleshooting
+
+If you found the federated resources were not propagated to the member clusters, you can always
```
Comment (marun) [inline docs/userguide.md]: s/If you found the federated resources were not propagated/If federated resources are not propagated as expected/
s/always//

File: docs/userguide.md [hunk 6]
```diff
@@ -305,6 +305,27 @@ To cleanup the example simply delete the namespace:
...
+use the following command to get some failure reason by checkig the `Events`.
+
+```bash
+kubectl describe <federate CRD> <CR> -n test-namespace
+```
+
+One example for CRD of `federatedserviceaccounts` is as follows:
```
Comment (marun) [inline docs/userguide.md]: s/One/An/

File: docs/userguide.md [hunk 7]
```diff
@@ -305,6 +305,27 @@ To cleanup the example simply delete the namespace:
...
+
+```bash
+kubectl describe federatedserviceaccounts test-serviceaccount -n test-namespace
+```
+
+If you cannot get too much help info by above command, you can check federation controller log as follows:
```
Comment (marun) [inline docs/userguide.md]: s/.../It may also be useful to inspect the federation controller log as follows:

**Comment by marun** [comment]
Much appreciated!

/approve
/lgtm

**Pull request merged**